src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4158,20 +4178,19 @@ private bool HasTopLevelNullabilityConversion(TypeWithAnnotations source, TypeWi
             bool extensionMethodThisArgument = false,
             Optional<LocalState> stateForLambda = default,
             bool trackMembers = false,
-            Location location = null)
+            Location diagnosticLocationOpt = null)
         {
             Debug.Assert(!trackMembers || !IsConditionalState);
-            Debug.Assert(node != null);
-            Debug.Assert(operandOpt != null || !operandType.HasNullType);
+            Debug.Assert(convertedNode != null);
             Debug.Assert(targetTypeWithNullability.HasType);
             Debug.Assert((object)target != null || assignmentKind != AssignmentKind.Argument);
 
             NullableFlowState resultState = NullableFlowState.NotNull;
             bool canConvertNestedNullability = true;
             bool isSuppressed = false;
-            location ??= node.Syntax.GetLocation();
+            diagnosticLocationOpt ??= (conversionOpt ?? convertedNode).Syntax.GetLocation();
 
-            if (operandOpt?.IsSuppressed == true)
+            if (convertedNode?.IsSuppressed == true)
```

Comment:
>if (convertedNode?.IsSuppressed == true) [](start = 12, length = 40)

The name of the parameter (`convertedNode`) threw me off here. Sounds like it's the "converted node", ie. after conversion rather  than the "node getting converted". I think "operand" was better.